extended/src/test/java/apoc/util/ExtendedUtilTest.java

```diff
@@ -0,0 +1,105 @@
+
+        assertEquals(4, result);
+
+        // The method will attempt to execute the operation with a linear backoff strategy,
+        // sleeping for 1 second, 2 seconds, and 3 seconds between retries.
+        // This results in a total wait time of 6 seconds (1s + 2s + 3s) if the operation succeeds on the third attempt,
+        // leading to an approximate execution time of 6 seconds.
+        assertTrue(time > 5500);
+        assertTrue(time < 6500);
+    }
+
+    @Test
+    public void testWithExponentialBackOffRetriesWithSuccess() {
+        i=0;
+        long start = System.currentTimeMillis();
+        int result = ExtendedUtil.withBackOffRetries(
+                this::testFunction,
+                true, 0, // test backoffRetry default value -> 5
+                runEx -> {
+                    if(!runEx.getMessage().contains("Expected"))
+                        throw new RuntimeException("Some Bad News...");
+                },
+                Boolean.TRUE
+        );
+        long time = System.currentTimeMillis() - start;
+
+        assertEquals(4, result);
+
+        // The method will attempt to execute the operation with an exponential backoff strategy,
+        // sleeping for 2 second, 4 seconds, and 8 seconds between retries.
+        // This results in a total wait time of 14 seconds (2s + 4s + 8s) if the operation succeeds on the third attempt,
+        // leading to an approximate execution time of 14 seconds.
+        assertTrue(time > 13500);
+        assertTrue(time < 14500);
+    }
+
+    @Test
+    public void testBackOffRetriesWithError() {
+        i=0;
+        long start = System.currentTimeMillis();
+        assertThrows(
+                RuntimeException.class,
+                () -> ExtendedUtil.withBackOffRetries(
+                        this::testFunction,
+                        true, 2,
+                        runEx -> {}
+                )
+        );
+        long time = System.currentTimeMillis() - start;
+
+        // The method is configured to retry the operation twice.
+        // So, it will make two extra-attempts, waiting for 1 second and 2 seconds before failing and throwing an exception.
+        // Resulting in an approximate execution time of 3 seconds.
+        assertTrue(time > 2500);
+        assertTrue(time < 3500);
+    }
+
+    @Test
+    public void testWithoutBackOffRetriesWithError() {
+        i=0;
+        assertThrows(
+                RuntimeException.class,
+                () -> ExtendedUtil.withBackOffRetries(
+                        this::testFunction,
+                        false, 30,
+                        runEx -> {}
+                )
+        );
+
+        // Retry strategy is not active and the testFunction is executed only once by raising an exception.
+        assertEquals(1, i);
+    }
+
+    private int testFunction() {
+        if (++i % 4 != 0) {
+            throw new RuntimeException("Expected i");
+        }
```

Comment:
Se ho capito bene quello che fa, lo scriverei così
```suggestion
       i++;
        if (i == 4) {
            throw new RuntimeException("Expected i not equal to 4");
        }
```